package proto_ff;

import "yd_fieldoptions.proto";


message E_MapMap
{
	optional int64 m_mapid = 1[(yd_fieldoptions.field_cname) = "地图id"];
	optional int32 m_levellimit = 2[(yd_fieldoptions.field_cname) = "地图等级限制"];
	optional string m_advicelv = 3[(yd_fieldoptions.field_cname) = "地图推荐等级段", (yd_fieldoptions.field_bufsize) = 60];
	optional int32 m_maptype = 4[(yd_fieldoptions.field_cname) = "地图类型"];
	optional int32 m_mapsubtype = 5[(yd_fieldoptions.field_cname) = "地图子类型"];
	optional int32 m_clearancetime = 6[(yd_fieldoptions.field_cname) = "地图清场时间（秒）"];
	optional int32 m_mappkmode = 7[(yd_fieldoptions.field_cname) = "地图pk模式"];
	optional int32 m_pklock = 8[(yd_fieldoptions.field_cname) = "地图是否允许切换PK模式"];
	optional int32 m_pk = 9[(yd_fieldoptions.field_cname) = "地图是否允许pk"];
	optional int32 m_killvalue = 10[(yd_fieldoptions.field_cname) = "地图杀人是否增加杀戮值且地图死亡后是否清除杀戮值"];
	optional int32 m_bangroup = 11[(yd_fieldoptions.field_cname) = "禁止组队进入邀请"];
	optional int32 m_revivetime = 12[(yd_fieldoptions.field_cname) = "场景复活倒计时长"];
	optional int32 m_mapwide = 13[(yd_fieldoptions.field_cname) = "地图宽度"];
	optional int32 m_mapheight = 14[(yd_fieldoptions.field_cname) = "地图高度"];
	optional string m_mapresources = 15[(yd_fieldoptions.field_cname) = "地图资源名", (yd_fieldoptions.field_bufsize) = 60];
	optional int64 m_reviveskill = 16[(yd_fieldoptions.field_cname) = "复活无敌"];
	optional int32 m_reviveinplacetype = 17[(yd_fieldoptions.field_cname) = "原地复活类型"];
	optional int32 m_ishangup = 18[(yd_fieldoptions.field_cname) = "是否能自动挂机"];
	optional int32 m_itemdroptype = 19[(yd_fieldoptions.field_cname) = "死亡掉落类型"];
	optional string m_itemdropid = 20[(yd_fieldoptions.field_cname) = "触发掉落组ID", (yd_fieldoptions.field_bufsize) = 60];
	optional int64 m_linkoffline = 21[(yd_fieldoptions.field_cname) = "下线后回到地图"];
	optional int32 m_banonhook = 22[(yd_fieldoptions.field_cname) = "是否禁止原地挂机"];
	optional int32 m_rewait = 23[(yd_fieldoptions.field_cname) = "普通复活等待时间"];
	optional int32 m_retime = 24[(yd_fieldoptions.field_cname) = "自动普通复活倒计时"];
	optional int32 m_hreban = 25[(yd_fieldoptions.field_cname) = "禁止原地复活"];
	optional int32 m_retiered = 26[(yd_fieldoptions.field_cname) = "是否疲劳复活"];
	optional int32 m_reskillid = 27[(yd_fieldoptions.field_cname) = "复活后获得的技能"];
	optional int32 m_reteimes = 28[(yd_fieldoptions.field_cname) = "复活次数限制"];
	optional int32 m_drivepoint = 29[(yd_fieldoptions.field_cname) = "驱赶值"];
	optional int32 m_diedp = 30[(yd_fieldoptions.field_cname) = "死亡增加驱赶值"];
	optional int32 m_timedp = 31[(yd_fieldoptions.field_cname) = "每分钟增加驱赶值"];
	optional int32 m_meditation = 32[(yd_fieldoptions.field_cname) = "是否允许打坐"];
}


message Sheet_MapMap
{
	repeated E_MapMap E_MapMap_List = 1[(yd_fieldoptions.field_arysize)=40];
}

message E_MapItemdrop
{
	optional int32 m_itemdropid = 1[(yd_fieldoptions.field_cname) = "掉落ID"];
	optional int32 m_droptype = 2[(yd_fieldoptions.field_cname) = "掉落类型"];
	optional int32 m_odds = 3[(yd_fieldoptions.field_cname) = "触发几率"];
	optional int32 m_dropvaluea = 4[(yd_fieldoptions.field_cname) = "掉落参数a"];
	optional int32 m_dropvalueb = 5[(yd_fieldoptions.field_cname) = "掉落参数b"];
	optional int32 m_dropvaluec = 6[(yd_fieldoptions.field_cname) = "掉落参数c"];
	optional int32 m_dropvalued = 7[(yd_fieldoptions.field_cname) = "掉落参数d"];
	optional int32 m_dropvaluee = 8[(yd_fieldoptions.field_cname) = "掉落参数e"];
	optional int32 m_dropvaluef = 9[(yd_fieldoptions.field_cname) = "掉落参数f"];
}


message Sheet_MapItemdrop
{
	repeated E_MapItemdrop E_MapItemdrop_List = 1[(yd_fieldoptions.field_arysize)=40];
}

message E_MapDestroyitem
{
	optional int64 m_destroyitemid = 1[(yd_fieldoptions.field_cname) = "触发摧毁道具ID"];
}


message Sheet_MapDestroyitem
{
	repeated E_MapDestroyitem E_MapDestroyitem_List = 1[(yd_fieldoptions.field_arysize)=200];
}
